pkg/backends/datadog/datadog.go

```diff
@@ -344,6 +364,19 @@ func (d *Client) authenticatedURL(path string) string {
 	return fmt.Sprintf("%s%s?%s", d.apiEndpoint, path, q.Encode())
 }
 
+func (d *Client) tsBorrow() *timeSeries {
+	return <-d.tsPool
+}
+
+func (d *Client) tsReturn(ts *timeSeries) {
+	// Zero the slice's contents to remove references to strings so that GC can collect them
+	for i := range ts.Series {
+		ts.Series[i] = metric{}
+	}
+	ts.Series = ts.Series[:0]
```

Comment:
372-375 shouldn't be necessary with this.  Data after the slice length should be GC'd unless there's another slice with a larger view of the data.

Reply:
GC does not understand that this data is not reachable via this slice (and also what if you re-slice it? It will become reachable). GC looks at the backing array and sees pointers and keeps their targets reachable.

https://github.com/golang/go/wiki/SliceTricks#delete-without-preserving-order

Reply:
That is many levels of horrifying.